xbmc/cores/AudioEngine/CMakeLists.txt

```diff
@@ -20,8 +20,7 @@ set(SOURCES AEResampleFactory.cpp
             Utils/AELimiter.cpp
             Utils/AEPackIEC61937.cpp
             Utils/AEStreamInfo.cpp
-            Utils/AEUtil.cpp
```

Comment:
This broke all sinks by accident, cause now no system.h was included anymore

Reply:
the one below this comment (obviously) -> therefore all the ifdefs ended up negative

Reply:
interesting, why did jenkins builds not fail?

Reply:
because it the function was still defined but it was the wrong function I guess

Reply:
what function? there is obviously something wrong if it build without this file

Reply:
We talk about: AESinkNULL.cpp not AEUtils.cpp see my comment: "the one below this comment".

Reply:
AESinkNULL.cpp included system.h which then was available for all the others. Now as it is not there anymore all the ifdefs in the OptionalReg.cpp file evaluate to false and no sink is registered anymore.

Reply:
if AEUtil is only used by sinks, we should move it to sinks folder

Reply:
Not the case:
```
xbmc/Application.cpp:#include "cores/AudioEngine/Utils/AEUtil.h"
xbmc/GUIInfoManager.cpp:#include "cores/AudioEngine/Utils/AEUtil.h"
xbmc/addons/AudioDecoder.cpp:#include "cores/AudioEngine/Utils/AEUtil.h"
xbmc/cores/AudioEngine/Encoders/AEEncoderFFmpeg.cpp:#include "cores/AudioEngine/Utils/AEUtil.h"
xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp:#include "cores/AudioEngine/Utils/AEUtil.h"
xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEBuffer.cpp:#include "cores/AudioEngine/Utils/AEUtil.h"
xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEResampleFFMPEG.cpp:#include "cores/AudioEngine/Utils/AEUtil.h"
xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEResamplePi.cpp:#include "cores/AudioEngine/Utils/AEUtil.h"
xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAESink.cpp:#include "cores/AudioEngine/Utils/AEUtil.h"
xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEStream.cpp:#include "cores/AudioEngine/Utils/AEUtil.h"
xbmc/cores/AudioEngine/Engines/ActiveAE/AudioDSPAddons/ActiveAEDSP.cpp:#include "cores/AudioEngine/Utils/AEUtil.h"
xbmc/cores/AudioEngine/Engines/ActiveAE/AudioDSPAddons/ActiveAEDSPProcess.cpp:#include "cores/AudioEngine/Utils/AEUtil.h"
xbmc/cores/AudioEngine/Interfaces/AESink.h:#include "cores/AudioEngine/Utils/AEUtil.h"
xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp:#include "cores/AudioEngine/Utils/AEUtil.h"
xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.cpp:#include "cores/AudioEngine/Utils/AEUtil.h"
xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.h:#include "cores/AudioEngine/Utils/AEUtil.h"
xbmc/cores/AudioEngine/Sinks/AESinkDARWINIOS.mm:#include "cores/AudioEngine/Utils/AEUtil.h"
xbmc/cores/AudioEngine/Sinks/AESinkDirectSound.cpp:#include "cores/AudioEngine/Utils/AEUtil.h"
xbmc/cores/AudioEngine/Sinks/AESinkOSS.cpp:#include "cores/AudioEngine/Utils/AEUtil.h"
xbmc/cores/AudioEngine/Sinks/AESinkPULSE.h:#include "cores/AudioEngine/Utils/AEUtil.h"
xbmc/cores/AudioEngine/Sinks/AESinkPi.cpp:#include "cores/AudioEngine/Utils/AEUtil.h"
xbmc/cores/AudioEngine/Sinks/AESinkSNDIO.cpp:#include "cores/AudioEngine/Utils/AEUtil.h"
xbmc/cores/AudioEngine/Sinks/AESinkWASAPI.cpp:#include "cores/AudioEngine/Utils/AEUtil.h"
xbmc/cores/AudioEngine/Sinks/AESinkXAudio.cpp:#include "cores/AudioEngine/Utils/AEUtil.h"
xbmc/cores/AudioEngine/Sinks/osx/AEDeviceEnumerationOSX.cpp:#include "cores/AudioEngine/Utils/AEUtil.h"
xbmc/cores/AudioEngine/Sinks/test/TestAESinkDARWINOSX.cpp:#include "cores/AudioEngine/Utils/AEUtil.h"
xbmc/cores/AudioEngine/Sinks/windows/AESinkFactoryWin.h:#include "cores/AudioEngine/Utils/AEUtil.h"
xbmc/cores/AudioEngine/Utils/AEDeviceInfo.cpp:#include "AEUtil.h"
xbmc/cores/AudioEngine/Utils/AEUtil.cpp:#include "AEUtil.h"
xbmc/cores/RetroPlayer/RetroPlayerAudio.cpp:#include "cores/AudioEngine/Utils/AEUtil.h"
xbmc/cores/VideoPlayer/DVDCodecs/Audio/DVDAudioCodec.h:#include "cores/AudioEngine/Utils/AEUtil.h"
xbmc/cores/VideoPlayer/DVDCodecs/Audio/DVDAudioCodecFFmpeg.cpp:#include "cores/AudioEngine/Utils/AEUtil.h"
xbmc/cores/VideoPlayer/VideoPlayerAudio.cpp:#include "cores/AudioEngine/Utils/AEUtil.h"
xbmc/cores/omxplayer/OMXAudio.h:#include "cores/AudioEngine/Utils/AEUtil.h"
xbmc/cores/omxplayer/OMXAudioCodecOMX.cpp:#include "cores/AudioEngine/Utils/AEUtil.h"
xbmc/cores/paplayer/PAPlayer.cpp:#include "cores/AudioEngine/Utils/AEUtil.h"
xbmc/cores/paplayer/VideoPlayerCodec.cpp:#include "cores/AudioEngine/Utils/AEUtil.h"
xbmc/settings/dialogs/GUIDialogAudioDSPSettings.cpp:#include "cores/AudioEngine/Utils/AEUtil.h"
xbmc/video/dialogs/GUIDialogAudioSubtitleSettings.cpp:#include "cores/AudioEngine/Utils/AEUtil.h"
```

AEUtils is perfectly fine :-) Again: AESinkNULL included system.h and made it available for everyone in scope of AE - nothing to do with AEUtils.

Reply:
The header with its definition might be used by other components but the body seems to be only used by sinks. 

> AEUtils is perfectly fine :-)

definitely not as long there is crap like CAESpinSection in this file

Reply:
Fernet :-) I will add "in the above regard" next time :-) I am also not sure if AEUtils affects "world peace" in a certain way.